client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/HeaderDecoder.java

```diff
@@ -90,11 +90,6 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
                   case CACHE_ENTRY_EXPIRED_EVENT_RESPONSE:
                      if (codec.allowOperationsAndEvents()) {
                         operation = messageId == 0 ? null : incomplete.get(messageId);
-                     } else if (incomplete.size() == 1) {
-                        operation = incomplete.values().iterator().next();
-                        messageId = operation.header().messageId();
-                     } else if (incomplete.size() > 1) {
-                        throw new IllegalStateException("Too many incomplete operations: " + incomplete);
```

Comment:
Adding this back and running the `ClientListenerFailoverBusyTest` will trigger the issues.